File: app/src/main/java/com/binge/easysubway/widget/Fab.java
package com.binge.easysubway.widget;

import android.content.Context;
import android.support.design.widget.FloatingActionButton;
import android.util.AttributeSet;

import com.gordonwong.materialsheetfab.AnimatedFab;

/**
 * Created by Binge on 2017/9/7.
 */

public class Fab extends FloatingActionButton implements AnimatedFab {
    public Fab(Context context) {
        super(context);
    }

    public Fab(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public Fab(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void show() {
        super.show();
        show(0,0);
    }

    @Override
    public void show(float translationX, float translationY) {
        setVisibility(VISIBLE);
    }
}

File: app/src/main/java/com/binge/easysubway/widget/Fab.kt
package com.binge.easysubway.widget

import android.content.Context
import android.support.design.widget.FloatingActionButton
import android.util.AttributeSet
import android.view.View
import com.gordonwong.materialsheetfab.AnimatedFab

/**
 * Created by Administrator on 2017/9/4.
 */
class Fab : FloatingActionButton, AnimatedFab {
    constructor(context: Context?) : super(context)
    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    override fun show() {
        show(0f, 0f)
    }

    override fun show(translationX: Float, translationY: Float) {
        visibility = View.VISIBLE
    }


}